web/sharings/sharings.go

```diff
@@ -413,35 +418,73 @@ func deleteDocument(c echo.Context) error {
 
 // Set sharing to revoked and delete all associated OAuth Clients.
 func revokeSharing(c echo.Context) error {
-	instance := middlewares.GetInstance(c)
+	ins := middlewares.GetInstance(c)
 
-	sharingID := c.Param("id")
-	sharing, err := sharings.FindSharing(instance, sharingID)
+	sharingID := c.Param("sharing-id")
+	sharing, err := sharings.FindSharing(ins, sharingID)
 	if err != nil {
 		return jsonapi.NotFound(err)
 	}
 
-	// TODO Add permission check
+	sharerID := ""
+	if sharing.Sharer.SharerStatus != nil {
+		sharerID = sharing.Sharer.SharerStatus.HostClientID
+	}
+
+	err = checkRevokeSharingPermissions(c, ins, sharing, sharerID)
+	if err != nil {
+		return c.JSON(http.StatusForbidden, err)
+	}
+
+	recursiveRaw := c.QueryParam(consts.QueryParamRecursive)
+	var recursive bool
+	if recursiveRaw == "" {
+		recursive = true
```

Comment:
[nitpick] I'd prefer to shorten the code a bit to make it explicit that `recursive` is `true` by default:

```go
recursive := true
if recursiveRaw != "" {
    // ...
}
```